### PR TITLE
Make middleware issues more discoverable

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/BasicMiddleware/issues/46

@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNetCore.HttpOverrides/ForwardedHeadersMiddleware.cs [hunk 1]
```diff
@@ -185,7 +185,7 @@ public void ApplyForwarders(HttpContext context)
                     }
                     else if (_options.RequireHeaderSymmetry)
                     {
-                        _logger.LogDebug(4, $"Failed to parse forwarded host: {set.Host}");
+                        _logger.LogWarning(4, $"Failed to parse forwarded host: {set.Host}");
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.HttpOverrides/ForwardedHeadersMiddleware.cs]: Let's clarify the message too. These are symmetry failures due to missing values. No point in showing the null/empty value.

**Comment by BrennanConroy** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.HttpOverrides/ForwardedHeadersMiddleware.cs [hunk 2]
```diff
@@ -185,7 +185,7 @@ public void ApplyForwarders(HttpContext context)
                     }
                     else if (_options.RequireHeaderSymmetry)
                     {
-                        _logger.LogDebug(4, $"Failed to parse forwarded host: {set.Host}");
+                        _logger.LogWarning(4, $"Forwarded host is not present, this is required by {nameof(_options.RequireHeaderSymmetry)}");
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.HttpOverrides/ForwardedHeadersMiddleware.cs]: "Incorrect number of x-forwarded-proto header values, see RequireHeaderSymmetry."

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**